src/core/config/config_module.rs

```diff
@@ -15,13 +15,45 @@ use crate::core::rest::{EndpointSet, Unchecked};
 
 /// A wrapper on top of Config that contains all the resolved extensions and
 /// computed values.
-#[derive(Clone, Debug, Default, Setters, MergeRight)]
+#[derive(Clone, Debug, Default, Getters, MergeRight)]
 pub struct ConfigModule {
-    pub config: Config,
-    pub extensions: Extensions,
-    pub input_types: HashSet<String>,
-    pub output_types: HashSet<String>,
-    pub interface_types: HashSet<String>,
+    config: Config,
+    extensions: Extensions,
+    input_types: HashSet<String>,
+    output_types: HashSet<String>,
+    interface_types: HashSet<String>,
+}
+
+impl ConfigModule {
+    // recompute the values of `input_types`, `output_types` and `interface_types`
+    pub fn recompute_types(self) -> Self {
```

Comment:
Don't need this method. Instead just implement a `set_config` method that internally calls `ConfigModule::from` and recomputes everything.

Reply:
_ext_links_ uses this function to recompute the types and in it's signature it returns _ConfigModule_ if the _set_config_ consumes self we will end up with ownership rule violations, with my PR I'm trying  to avoid cloning by returning ownership to the caller. 

If you want we can create a method that's called _set_config_module_ that consumes self and returns the ownership of config_module back to the caller OR if you're ok with cloning we can go with that approach